### PR TITLE
Add DOM Events types

### PR DESCRIPTION
**Pull request by sorta-shopify**
### Background
https://github.com/Shopify/ce-customer-behaviour/issues/3379

### Solution
Updating our package with the latest types, which will include the types from DOM Events.

### 🎩

- ...

### Checklist

- [ ] I have :tophat:'d these changes
- [x] I have updated relevant documentation

### CODE REVIEW

**Review by h74zhou — APPROVED** [review]
Looks good!!!

File: packages/web-pixels-extension/src/schemas/pixel-events.ts [hunk 1]
```diff
@@ -453,6 +453,41 @@ export const pixelEvents = {
         },
       },
     },
+    EventType: {
+      enum: ['standard', 'dom', 'custom', 'meta'],
```
[the remark this answers is not in the document]
Comment (sorta-shopify) [reply packages/web-pixels-extension/src/schemas/pixel-events.ts:457]: This is making me think, maybe we need two different types. A public EventType, and an internal/private equivalent. meta is really only needed for subscribe, to allow things like `all_standard_events` etc.

File: packages/web-pixels-extension/src/types/PublicApi.ts [hunk 2]
```diff
@@ -4,5 +4,6 @@ import type {VisitorApi} from './VisitorApi';
 export interface PublicApi {
   publish: EventBus['publish'];
   publishCustomEvent: EventBus['publishCustomEvent'];
+  subscribe: EventBus['subscribe'];
   visitor: VisitorApi['visitor'];
 }
```
[the remark this answers is not in the document]
Comment (sorta-shopify) [reply packages/web-pixels-extension/src/types/PublicApi.ts:9]: Hmm, I don't think I did this as part of DOM Events. I vaguely remember someone doing something about exposing `subscribe` in certain contexts 🤔

[the remark this answers is not in the document]
Comment (antoinegrant) [reply packages/web-pixels-extension/src/types/PublicApi.ts:9]: I did expose the `subscribe`. It's by design, but I'm more concerned with exposing that interface. Maybe we can revisit that at some point.

File: packages/web-pixels-extension/src/schemas/pixel-events.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sorta-shopify) [reply packages/web-pixels-extension/src/schemas/pixel-events.ts:457]: Ok cool, sounds good 👍

**Pull request merged**